doc: add instructions for runing unit tests (#4576)
pingcap/tiflash: README.md

## Run Unit Tests

To run unit tests, you need to build with `-DCMAKE_BUILD_TYPE=DEBUG`:

```shell
cd $BUILD

> cmake $WORKSPACE/tiflash -GNinja -DCMAKE_BUILD_TYPE=DEBUG

Reviewer: ```suggestion
cmake $WORKSPACE/tiflash -GNinja -DCMAKE_BUILD_TYPE=DEBUG -DENABLE_TESTS=ON
```

Author: https://github.com/pingcap/tiflash/blob/master/CMakeLists.txt#L379
```cmake
# Enable tests by default when build type is debug
if (CMAKE_BUILD_TYPE_UC STREQUAL "DEBUG")
    set (ENABLE_TESTS_DEFAULT ON)
else ()
    set (ENABLE_TESTS_DEFAULT OFF)
endif ()
```

Reviewer: got it, thx

ninja gtests_dbms
```

> And the executable unit test is in `$BUILD/dbms/gtests_dbms`.

Reviewer: better to mention `gtests_libcommon` too.

Author: ok, I will fix

## Run Unit Tests

To run unit tests, you need to build with `-DCMAKE_BUILD_TYPE=DEBUG`:

```shell
cd $BUILD
cmake $WORKSPACE/tiflash -GNinja -DCMAKE_BUILD_TYPE=DEBUG
ninja gtests_dbms
ninja gtests_libcommon
ninja gtests_libdaemon
```

> And the executable unit test is in `$BUILD/dbms/gtests_dbms`, `$BUILD/libs/libcommon/src/tests/gtests_libcommon` and `$BUILD/libs/libdaemon/src/tests/gtests_libdaemon`.

Reviewer 2: ```suggestion
And the unit-test executables are at `$BUILD/dbms/gtests_dbms`, `$BUILD/libs/libcommon/src/tests/gtests_libcommon` and `$BUILD/libs/libdaemon/src/tests/gtests_libdaemon`.
```
